File: frontend/src/app/models/project.ts
import {User} from "./user";

export class Project {
  constructor(public id: string,
              public name: string,
              public description: string,
              public createdAt: Date,
              public deadLine: Date,
              public imageUrls: string[],
              public overallBack: number,
              public currentBack: number,
              public author: User) {
  }
}

File: user/src/main/java/com/od/crowdy/user/domain/neo4j/model/Project.java
package com.od.crowdy.user.domain.neo4j.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.neo4j.springframework.data.core.schema.GeneratedValue;
import org.neo4j.springframework.data.core.schema.Node;
import org.neo4j.springframework.data.core.support.UUIDStringGenerator;
import org.springframework.data.annotation.Id;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Node("Project")
public class Project {

    @Id
    @GeneratedValue(UUIDStringGenerator.class)
    private String id;

    private String name;
    private String description;
    private LocalDate createdAt;
    private LocalDate deadline;
    private List<String> imageUrls;
    private BigDecimal overallBack;
    private BigDecimal currentBack;
}
